x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunnerTests.java

```diff
@@ -259,14 +259,15 @@ public void testRunStateChangePolicyWithNextStep() throws Exception {
 
         ClusterState before = clusterService.state();
         CountDownLatch latch = new CountDownLatch(1);
-        step.setLatch(latch);
+        nextStep.setLatch(latch);
         runner.runPolicyAfterStateChange(policyName, indexMetaData);
 
-        latch.await(5, TimeUnit.SECONDS);
-        ClusterState after = clusterService.state();
+        assertTrue(latch.await(5, TimeUnit.SECONDS));
 
-        assertEquals(before, after);
+        // The cluster state can take a few extra milliseconds to update after the steps are executed
+        assertBusy(() -> assertNotEquals(before, clusterService.state()));
         assertThat(step.getExecuteCount(), equalTo(1L));
+        assertThat(nextStep.getExecuteCount(), equalTo(1L));
```

Comment:
I think we should also check that the clusterstate shows the correct ILM state at this point too

Reply:
Done.  There's a few other tests in this class that could probably use the same treatment, but that's a separate PR.